Eutherion.Text.Json/Eutherion.Text/Json/JsonKeyValueSyntax.Green.cs

```diff
@@ -29,21 +29,6 @@ namespace Eutherion.Text.Json
     /// </summary>
     public sealed class GreenJsonKeyValueSyntax : ISpan
```

Comment:
Consider making all Green syntax nodes internal now there's no point in exposing their members in an API separately from regular syntax nodes.

Reply:
Can also remove argument checks and summaries for constructors that have become internal.

Reply:
Postpone until later. The current structure allows for other parsers to be written as long as they construct the same syntax tree. Reason I wanted to make it internal is because by adding the source json to RootJsonSyntax, it loses a bit of integrity, since it may not match with the generated parse tree, which was previously always the case - or rather, the responsibility of the caller code. Plus, of what use are green syntax nodes to consumer code anyway after this API roundup?

Another consideration is that there's a use case for parsing from a TextReader. In such cases, a source json is not even available. I added the Json property because it allows JsonSyntax nodes to lazily derive information by creating substrings of the source json after they've been created. So this is conceptually likely to be the fastest option - provided that the entire json string already existed in memory before the parse even started, as is the case with text editors.

So I'll postpone this kind of thing until this use case (and possibly also parsing from a ReadOnlySpan<char> or IEnumerable<char> if I feel particularly crazy) are implemented. Then it will also become clear what to do with that Json field.

I may opt for different types of root syntax nodes depending on how they were created and hide them behind an interface. The interface should then expose a method for providing source strings to syntax nodes. Or, hide the json property itself behind an indirection.

This also opens the option to create an implementation which simply throws away all such information except the relevant created json value, and then throws an exception if someone tries to examine what the comment text was in the source json.

Musings for a far future, if ever. Until then, better just accept the risk and have a string to refer to.